orttraining/orttraining/python/training/ortmodule/_graph_execution_manager.py

```diff
@@ -285,22 +289,43 @@ def _get_session_config(self):
 
         providers = None
         provider_options = None
-        if self._device.type == "cuda":
+        if self._device.type == 'cuda':
+            providers = []
+            provider_options = []
+
             # Configure the InferenceSessions to use the specific GPU on which the model is placed.
-            providers = ["ROCMExecutionProvider"] if self.is_rocm_pytorch else ["CUDAExecutionProvider"]
-            providers.append("CPUExecutionProvider")
-            provider_option_map = {"device_id": str(self._device.index)}
-            if not self.is_rocm_pytorch:
-                # Set Conv algo search mode to HEURISTIC, which is same as PyTorch's default setting.
-                provider_option_map["cudnn_conv_algo_search"] = "HEURISTIC"
-                provider_option_map["cudnn_conv_use_max_workspace"] = "1"
-                provider_option_map["cudnn_conv1d_pad_to_nc1d"] = "1"
+            cuda_or_rocm_provider_option_map = {"device_id": str(self._device.index)}
             if self._use_external_gpu_allocator:
-                provider_option_map["gpu_external_alloc"] = str(self._torch_alloc)
-                provider_option_map["gpu_external_free"] = str(self._torch_free)
-                provider_option_map["gpu_external_empty_cache"] = str(self._torch_empty_cache)
-            provider_options = [provider_option_map, {}]
-        elif self._device.type == "cpu":
+                cuda_or_rocm_provider_option_map["gpu_external_alloc"] = str(self._torch_alloc)
+                cuda_or_rocm_provider_option_map["gpu_external_free"] = str(self._torch_free)
+                cuda_or_rocm_provider_option_map["gpu_external_empty_cache"] = str(self._torch_empty_cache)
+
+            if self.is_rocm_pytorch:
+                providers.append("ROCMExecutionProvider")
+                provider_options.append(cuda_or_rocm_provider_option_map)
+            else:
+                if self._use_tensorrt_backend:
+                    providers.append("TensorrtExecutionProvider")
+                    # TODO: pass current torch stream into backend.
+                    # Currently, fix the compute stream onto default stream, aligned with CUDA EP.
```

Comment:
Why is only default stream used? TRT can use any stream as long as it's same as CUDA EP.

Reply:
We had an assumption in context of ORTModule, PyTorch model code also use default steam as the compute stream. ORT training (CUDA EP) use default stream as the compute stream.

So there won't be stream sync issue when there are interop between PyTorch and ORT (for example, PyTorch prepare inputs and call ORT to run the forward graph, ORT returned forward results; PyTorch use the results as inputs do other computations). 

When we enable TensorRT, when TRT EP creation, a non-default stream is created; also here https://github.com/microsoft/onnxruntime/blob/6cd1931a935b3694f027184f52abad1e8064f571/onnxruntime/core/session/inference_session.cc#L508, we tried to set CUDA EP using the same stream of TRT EP.

So there is stream sync issue in the PyTorch/ORT interop boundary, (and we did not handle the stream sync explicitly any more now).